providers/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -692,6 +737,12 @@ def read_pod_logs(
             post_termination_timeout=post_termination_timeout,
         )
 
+    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
+    def get_init_container_names(self, pod: V1Pod) -> list[str]:
+        """Return container names from the POD except for the airflow-xcom-sidecar container."""
```

Comment:
in every new method, please add the `:meta private:` sphinx directive.  this makes it clear it's not part of public interface.

Reply:
This method mirror `get_container_names`:

```
    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
    def get_init_container_names(self, pod: V1Pod) -> list[str]:
        """Return container names from the POD except for the airflow-xcom-sidecar container."""
        pod_info = self.read_pod(pod)
        return [container_spec.name for container_spec in pod_info.spec.init_containers]

    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
    def get_container_names(self, pod: V1Pod) -> list[str]:
        """Return container names from the POD except for the airflow-xcom-sidecar container."""
        pod_info = self.read_pod(pod)
        return [
            container_spec.name
            for container_spec in pod_info.spec.containers
            if container_spec.name != PodDefaults.SIDECAR_CONTAINER_NAME
        ]
```

In this case `get_container_names` also will require same logic.

Reply:
those methods are already there.  you are adding a new method.  just because we didn't do something in the past that we probably should have, it doesn't follow that we shouldn't do the good thing now.

Reply:
Fixed

Reply:
Fixed